types/underscore/underscore-tests.ts

```diff
@@ -1116,6 +1047,141 @@ declare const extractChainTypes: ChainTypeExtractor;
     _.chain([{a: 'a'}, {a: 'b'}]).findLastIndex({ a: 'b' }).value(); // $ExpectType number
 }
 
+// Objects
+
+// isEqual
+{
+    _.isEqual(anyValue, anyValue); // $ExpectType boolean
+    _(anyValue).isEqual(anyValue); // $ExpectType boolean
+    extractChainTypes(_.chain(anyValue).isEqual(anyValue)); // $ExpectType ChainType<boolean, never>
+}
+
+// isEmpty
+{
+    _.isEmpty(anyValue); // $ExpectType boolean
+    _(anyValue).isEmpty(); // $ExpectType boolean
+    extractChainTypes(_.chain(anyValue).isEmpty()); // $ExpectType ChainType<boolean, never>
+}
+
+// isMatch
+{
+    _.isMatch(anyValue, anyValue); // $ExpectType boolean
+    _(anyValue).isMatch(anyValue); // $ExpectType boolean
+    extractChainTypes(_.chain(anyValue).isMatch(anyValue)); // $ExpectType ChainType<boolean, never>
+}
+
+// isElement
+{
+    _.isElement(anyValue) ? anyValue : neverValue; // $ExpectType Element
+    _(anyValue).isElement(); // $ExpectType boolean
+    extractChainTypes(_.chain(anyValue).isElement()); // $ExpectType ChainType<boolean, never>
+}
+
+// isArray
+{
+    _.isArray(anyValue) ? anyValue : neverValue; // $ExpectType any[]
+    _(anyValue).isArray(); // $ExpectType boolean
+    extractChainTypes(_.chain(anyValue).isArray()); // $ExpectType ChainType<boolean, never>
+}
+
+// isSymbol
+{
+    _.isSymbol(anyValue) ? anyValue : neverValue; // $ExpectType symbol
+    _(anyValue).isSymbol(); // $ExpectType boolean
+    extractChainTypes(_.chain(anyValue).isSymbol()); // $ExpectType ChainType<boolean, never>
+}
+
+// isObject
+{
+    _.isObject(anyValue) ? anyValue : neverValue; // $ExpectType object
+    _(anyValue).isObject(); // $ExpectType boolean
+    extractChainTypes(_.chain(anyValue).isObject()); // $ExpectType ChainType<boolean, never>
+}
+
+// isArguments
+{
+    _.isArguments(anyValue) ? anyValue : neverValue; // $ExpectType IArguments
+    _(anyValue).isArguments(); // $ExpectType boolean
+    extractChainTypes(_.chain(anyValue).isArguments()); // $ExpectType ChainType<boolean, never>
+}
+
+// isFunction
+{
+    _.isFunction(maybeFunction) ? maybeFunction : neverValue; // $ExpectType () => void
```

Comment:
Surprisingly for some reason using `anyValue` here does not end up changing the perceived type of `anyValue` to `Function`, so I thought it would be interesting to check something that was a more specific function type than `Function`.